docs/dev/README.md

```diff
@@ -0,0 +1,218 @@
+# Development guide for aws-crt-python
+
+This guide is for contributors to aws-crt-python's source code.
+Familiarity (but not necessarily expertise) with Python and C is assumed.
+
+### Table of Contents
+
+*   [Git](#git)
+*   [Set up a Virtual Environment](#set-up-a-virtual-environment)
+*   [Install](#install)
+*   [Run Tests](#run-tests)
+    *   [Environment Variables for Tests](#environment-variables-for-tests)
+*   [Using an IDE](#using-an-ide)
+    *   [Visual Studio Code](#using-visual-studio-code-vscode)
+        *   [Debugging Python](#debugging-python-with-vscode)
+        *   [Debugging C](#debugging-c-with-vscode)
+
+## Git
+
+Clone to a development folder:
+```sh
+$ git clone git@github.com:awslabs/aws-crt-python.git
+$ cd aws-crt-python
+$ git submodule update --init
+```
+
+Note that you MUST manually update submodules any time you pull latest, or change branches:
+```sh
+$ git submodule update
+```
+
+## Set up a Virtual Environment
+
+Set up a [virtual environment](https://docs.python.org/3/library/venv.html)
+for development. This guide suggests `aws-crt-python/.venv/` as a default location.
+Create a virtual environment like so:
+```sh
+$ python3 -m venv .venv/
+```
+
+To activate the virtual environment in your current terminal:
+```sh
+$ source .venv/bin/activate
+```
+
+Your terminal looks something like this when the virtual environment is active:
+```
+(.venv) $
+```
+Now any time you type `python3` or `pip` or even `python`, you are using the
+one from your virtual environment.
+To stop using the virtual environment, run `deactivate`.
+
+## Install
```

Comment:
ooc, does this install any updates only to the virtual environment you setup and activated above? If yes, should we move the installs/updates to before venv to make sure all this stuff is installed across the system? Maybe we don't care?

Reply:
Great suggestion though :|

Found a stackoverflow about this weirdness:
https://unix.stackexchange.com/questions/464671/why-does-pythons-pip-reset-to-version-10-0-1-in-every-new-virtual-environment